src/vue-document.js

```diff
@@ -106,6 +106,13 @@ export default function VueDocument (Vue, o) {
         this.$root[cache] = undefined
         injector.call(this, window.document)
       }
+    },
+    // not called on server-side
+    destroyed: function () {
+      if (this[prop]) {
+        this.$root[cache] = undefined
+        injector.call(this, window.document)
```

Comment:
I don't understand all of this code completely, but I'm pretty sure that this is the right way to call the injector. I don't know how the cache works so I was just copying what you did. It worked in my tests.